### PR TITLE
Features/grafana k6

### PR DESCRIPTION
**Pull request by bradhead**
# Fixes or Implements k6 to use API updates, and  upgrades to grafana k6

## Description

this updates k6 to use the grafana k6 and adds more tests for APIs and includes reading the configurations rather than hard-coded resource URIs.  This preps us to run tests on managed APIs (Kong)

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x]  Not Required

## UI Changes

none

## Notes

linking to [AB#13643](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/13643)

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Testing/k6/services/encounter/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# HG Patient Service Testing  of various test types.
```
Comment (MikeLyttle) [inline Testing/k6/services/encounter/README.md:1]: This seems to be copied from the readme for patient and the title hasn't been updated to describe encounters.

Similarly, /Testing/k6/services/immunization/README.md looks like it was copied from the readme for medications and the title hasn't been updated to describe immunizations.

File: Testing/k6/webapp/spa/README.md [hunk 2]
```diff
@@ -9,5 +9,10 @@ HG_ENV=test HG_KEY=apikey HG_PASSWORD=pwd docker compose up
 ### Options
 
 HG_KEY is the Akamai Queue-IT api key to bypass the queue.
-HG_ENV is a choice of dev, test, prod
-HG_PASSWORD is the password for the loadtest_0x keycloak accounts, where x is from 1 to 15.
\ No newline at end of file
+HG_ENV is a choice of dev, test   (for production use the URL setting)
+HG_URL overrides the default url that is build from the environment setting. Use this for testin
```
Comment (MikeLyttle) [inline Testing/k6/webapp/spa/README.md:13]: This sentence seems to trail off.

File: Testing/k6/webapp/timeline/README.md [hunk 3]
```diff
@@ -10,4 +10,6 @@ HG_ENV=test HG_KEY=apikey HG_PASSWORD=pwd docker compose up
 
 HG_KEY is the Akamai Queue-IT api key to bypass the queue.
 HG_ENV is a choice of dev, test, prod
+HG_TYPE is the text type to run.
```
Comment (MikeLyttle) [inline Testing/k6/webapp/timeline/README.md:13]: Typo: "text type" => "test type".

**Review by MikeLyttle — APPROVED** [review]

**Review by rayking29 — APPROVED** [review]

File: Testing/k6/services/encounter/README.md (hunk 1 above)
Comment (bradhead) [reply Testing/k6/services/encounter/README.md:1]: ooh good cactch will update.

**Review by MikeLyttle — APPROVED** [review]

**Review by rayking29 — APPROVED** [review]

**Pull request merged**